src/main/java/examples/NetExamples.java

```diff
@@ -550,4 +552,31 @@ public void example47(Vertx vertx) {
             .setUsername("username").setPassword("secret"));
     NetClient client = vertx.createNetClient(options);
   }
+
+  public void example48(Vertx vertx) throws CertificateException {
+    SelfSignedCertificate certificate = SelfSignedCertificate.create();
+
+    NetServerOptions serverOptions = new NetServerOptions()
+      .setSsl(true)
+      .setKeyCertOptions(certificate.keyCertOption())
+      .setTrustOptions(certificate.trustOptions());
+
+    NetServer server = vertx.createNetServer(serverOptions)
+      .connectHandler(socket -> socket.write("Hello!").end())
+      .listen(1234, "localhost");
+
+    NetClientOptions clientOptions = new NetClientOptions()
+      .setSsl(true)
+      .setKeyCertOptions(certificate.keyCertOption())
+      .setTrustOptions(certificate.trustOptions());
+
+    NetClient client = vertx.createNetClient(clientOptions);
+    client.connect(1234, "localhost", ar -> {
+      if (ar.succeeded()) {
+        ar.result().handler(System.out::println);
```

Comment:
should not use a method reference and instead a lambda because of the code generation in other languages for the doc